### ISSUE TITLE
[Accessibility]: Code Insights escape filter panel loses focus

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41820]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33499

### Problem description

![PixelSnap 2022-09-20 at 15 39 00](https://user-images.githubusercontent.com/1855233/191359866-e902722f-5b2d-438e-adba-e4d87ddc77c7.png)

Before expanding this filter panel the filter button has focus. When closing the filter panel by pressing `esc` all focus is lost. Tabbing from here starts at the browser nav bar and cycles through all browser tabs before returning to the page.

Shift-tabbing after closing the filter panel focuses on the last insight in the dashboard.

### Expected behavior

Return focus to the filter button.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @joelkw @felixfbecker @vovakulikov

**Comment by vovakulikov** [comment]
Closed by https://github.com/sourcegraph/sourcegraph/pull/43970
